Automattic/wp-calypso
# Add checkout plan upsell component on domain only flow

## Changes proposed in this Pull Request

This PR add a plan upsell component on checkout page, for domain only flow.

In order to do this, it creates a shadow blog BEFORE the checkout, passing the `is_domain_only` option to the new blog api.

![domain-only-plan-upsell](https://user-images.githubusercontent.com/2797601/162699152-df511171-acde-4227-aea7-58887dc7ab69.png)

## Testing instructions

- Build this branch locally or open the live Calypso link
- Go `/start/domain`
- Enter a domain name
- Select _Just buy a domain_
- Verify that in the checkout page the plan upsell component is rendered
- Complete the checkout (without adding a plan) and verify that domain is correctly purchased  (as domain-only domain)
